src/Ocelot.Provider.Polly/PollyCircuitBreakingDelegatingHandler.cs

```diff
@@ -26,9 +27,17 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
         {
             try
             {
-                return await Policy
-                    .WrapAsync(_qoSProvider.CircuitBreaker.Policies)
-                    .ExecuteAsync(() => base.SendAsync(request, cancellationToken));
+                var policies = _qoSProvider.CircuitBreaker.Policies;
+                if (policies.Any())
+                {
+                    IAsyncPolicy policy = policies.Length >= 2
+                        ? Policy.WrapAsync(policies)
+                        : policies[0];
```

Comment:
I would reverse this check to reduce cognitive complexity. First check for 0 with `policies.Any()`, then check for 1 and otherwise wrap the policies because there are more.

```suggestion
                    IAsyncPolicy policy = policies.Length == 1
                        ? policies[0]
                        : Policy.WrapAsync(policies);
```

Reply:
Good point! But... We have a deal with white box, not black box.
Your point about "cognitive complexity" is right for black boxes only.
So, we can review and even change source code of the feature, because we have the white box.
I am saying about concrete implementation, about the [PollyQoSProvider](../../blob/develop/src/Ocelot.Provider.Polly/PollyQoSProvider.cs#L14) class.
This class is injected to delegating handler in the `AddPolly` extension method, see [line 38](../../blob/develop/src/Ocelot.Provider.Polly/OcelotBuilderExtensions.cs#L38) plz.

Let's return back to `PollyQoSProvider` [line 57](../../blob/develop/src/Ocelot.Provider.Polly/PollyQoSProvider.cs#L57) and we will see that `CircuitBreaker` object is initialized with 2 policies.
**Finally**, we have two cases:
- There is no QoS feature enabled. **0** policies
- QoS feature is enabled and initialized by **2** policies

So, **0** or **2** policies.
I believe my code flow is correct. We check for **0** policies first by `Any`.
If the feature enabled we check for **2** policies first.
But in theory one policy can be defined, and this is the last operator in code flow.

Truly speaking, Tom forgot to wrap these 2 policies while creating circuit breaker in the `PollyQoSProvider` class. That's why we have this bug issue.
Finally, I would prefer to move wrapping logic to the `PollyQoSProvider` class. Let's see how Tom will review this PR.

Reply:
The order doesn't really matter. It would also be an improvement to just check for `> 1` instead of `>= 2`. I think this is more readable because _"more than 1"_ is more straight forward than _"more than 2 or exactly 2"_.

```suggestion
                    IAsyncPolicy policy = policies.Length > 1
                        ? Policy.WrapAsync(policies)
                        : policies[0];
```

Reply:
Fixed in commit [Fix code review issue: "more than 1"](https://github.com/ThreeMammals/Ocelot/pull/1279/commits/7b431bc8b8fc871cc1356cea8eeda9740da7adcd)